### PR TITLE
[ws-daemon] Improve path concatenation

### PR DESCRIPTION
**Pull request by aledbf**

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **aledbf** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/ws-daemon/pkg/content/tar.go [hunk 1]
```diff
@@ -86,7 +87,10 @@ func TarWithOptions(srcPath string, options *TarOptions) (io.ReadCloser, error)
 			}
 
 			if relFilePath != "." {
-				relFilePath = strings.Join([]string{".", relFilePath}, string(filepath.Separator))
+				buffer := bytes.NewBufferString(".")
```
Comment (csweichel) [inline components/ws-daemon/pkg/content/tar.go:90]: better? `filepath.Join`

Comment (aledbf) [reply components/ws-daemon/pkg/content/tar.go:90]: Not really
```shell
❯ go test -benchmem -run=^$ -bench . bench_test.go | prettybench
goos: linux
goarch: amd64
cpu: AMD Ryzen 9 5900X 12-Core Processor            
PASS
benchmark                                    iter     time/iter   bytes alloc        allocs
---------                                    ----     ---------   -----------        ------
BenchmarkNewBufferString-24              76683656   15.20 ns/op        0 B/op   0 allocs/op
BenchmarkNewBufferStringParallel/1-24   880934428    1.38 ns/op        0 B/op   0 allocs/op
BenchmarkNewBufferStringParallel/2-24   793567332    1.53 ns/op        0 B/op   0 allocs/op
BenchmarkNewBufferStringParallel/4-24   826919538    1.50 ns/op        0 B/op   0 allocs/op
BenchmarkNewBufferStringParallel/8-24   809302243    1.46 ns/op        0 B/op   0 allocs/op
BenchmarkFilePathJoin-24                 13993242   83.81 ns/op       40 B/op   3 allocs/op
BenchmarkFilePathJoinParallel/1-24       94575600   13.63 ns/op       40 B/op   3 allocs/op
BenchmarkFilePathJoinParallel/2-24       72334652   14.96 ns/op       40 B/op   3 allocs/op
BenchmarkFilePathJoinParallel/4-24       80168283   15.62 ns/op       40 B/op   3 allocs/op
BenchmarkFilePathJoinParallel/8-24       71974920   15.55 ns/op       40 B/op   3 allocs/op
ok  	command-line-arguments	12.709s
```

Comment (aledbf) [reply components/ws-daemon/pkg/content/tar.go:90]: To be clear, I am trying to tackle this issue
Screenshot from 2021-08-10 21-24-54

Comment (csweichel) [reply components/ws-daemon/pkg/content/tar.go:90]: wow, that's surprising. Thank's for the benchmark 🚀

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5143#issuecomment-896869998" title="LGTM">csweichel</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5143#issuecomment-896870232" title="Approved">csweichel</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/5143#issuecomment-896870232" title="Approved">csweichel</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**